src/app/options/options.component.spec.ts

```diff
@@ -97,4 +97,15 @@ describe('OptionsComponent', () => {
 
     expect(component.data.additionalOptions).toEqual("Some additional options text");
   });
+
+  it('should call service.submit when submit button is pressed', async () => {
+    await fixture.whenStable();
+    spyOn<any>(component['service'], 'submit');
+    const submit = fixture.debugElement.query(By.css('.submitButton')).nativeElement;
+
+    submit.click();
+    fixture.detectChanges();
+
+    expect(component['service'].submit).toHaveBeenCalled();
```

Comment:
What I'd like to see tested is based on user input, additional options input etc, when the submit route is called, that the response returned from the call contains the OpenBabelData observable object you expect.